src/main/java/com/internet/shop/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,46 @@
+package com.internet.shop.filter;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.annotation.WebInitParam;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebFilter(initParams =
+            @WebInitParam(name = "urls", value = "/home,/login,/register"),
+                urlPatterns = "/*")
+public class AuthenticationFilter implements Filter {
+    private static final String USER_ID = "userId";
+    private List<String> excludeUrls;
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        excludeUrls = Arrays.asList(filterConfig.getInitParameter("urls").split(","));
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String servletPath = req.getServletPath();
+        if (!excludeUrls.contains(servletPath)
+                && req.getSession().getAttribute(USER_ID) == null) {
+            resp.sendRedirect("/home");
```

Comment:
let's push user to login page if he want to go to other pages, being unauthorized

Reply:
I can't do it because I used modal for login page. and I can't make redirect to modal

Reply:
login controller?

Reply:
I have login controller without view for it. Just logic for login